### ISSUE TITLE
Non-primitive values should not be put into path parameters

### ISSUE THREAD

**Issue by diego92sigma6** [googleapis/gapic-generator-java#1032]
Similar to #1016 
_In some cases, the generated client library would put complex object to ~query~ path parameters, like [here](https://github.com/googleapis/gapic-generator-java/blob/889f316d7dfcfdaf826cb8024166cc8eabeaef0d/src/test/java/com/google/api/generator/gapic/composer/rest/goldens/HttpJsonComplianceStub.golden#L176) , the proto definition for this method is [here](https://github.com/googleapis/gapic-generator-java/blob/889f316d7dfcfdaf826cb8024166cc8eabeaef0d/src/test/proto/compliance.proto#L64). The generated client library will error out during runtime if non-primitive values are put into query parameters._

**Comment by suztomo** [comment]
Is this causing a problem somebody suffers from?

**Comment by blakeli0** [comment]
This is hypothesis scenario, we don't have a service proto that is configured this way. But if they do, the generated code would error out during runtime.

**Comment by suztomo** [comment]
My impression is that if such proto file is introduced it breaks most of the languages. Or is there a specification that GAPIC generator Java hasn't implemented yet?

**Comment by blakeli0** [comment]
I don't think we have an AIP or similar specification that forbid this scenario, the [comment](https://github.com/googleapis/googleapis/blob/dbfbfdb38a2f891384779a7ee31deb15adba6659/google/api/http.proto#L61-L65) in the proto is the closest thing. Yes you are probably right that it would most likely break other languages as well. Ideally this should be validated in a language neutral tool like APILinter as well.

**Comment by suztomo** [comment]
So the proper solution seems not to make the Java generator to work with such proto files (somehow), but to prevent such proto files.

**Comment by blakeli0** [comment]
> So the proper solution seems not to make the Java generator to work with such proto files (somehow), but to prevent such proto files.

Agreed. Maybe it was not clear, what proposed in this issue is to add validation in the Java generator so that the generator would error out instead of erroring out during runtime.
